src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs

```diff
@@ -495,6 +496,68 @@ End Class
 ");
         }
 
+        [Fact]
+        public async Task CA2200_DiagnosticsForThrowCaughtExceptionInLocalMethod()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    void CatchAndRethrowExplicitly()
+    {
+        try
+        {
+        }
+        catch (Exception e)
+        {
+
+            DoThrow();
+
+            void DoThrow()
+            {
+                [|throw e;|]
+            }
+        }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task CA2200_NoDiagnosticsForThrowCaughtExceptionInActionOrFunc()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+class Program
+{
+    void CatchAndRethrowExplicitly()
+    {
+        try
+        {
+        }
+        catch (Exception e)
+        {
+            Action DoThrowAction = () =>
+            {
+                throw e;
```

Comment:
Maybe we do want to report here as we do report for the inner throw in https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/RethrowToPreserveStackDetailsTests.cs#L324-L351

Reply:
I don't think we want to report for throw crossing method boundaries, this lambda can be invoked after say `e` has been re-assigned a different exception. That would be a false positive.